src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/LimitedUseItem.java

```diff
@@ -0,0 +1,149 @@
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.bukkit.persistence.PersistentDataContainer;
+import org.bukkit.persistence.PersistentDataType;
+
+import io.github.thebusybiscuit.cscorelib2.chat.ChatColors;
+import io.github.thebusybiscuit.slimefun4.api.SlimefunAddon;
+import io.github.thebusybiscuit.slimefun4.core.handlers.ItemUseHandler;
+import io.github.thebusybiscuit.slimefun4.implementation.SlimefunPlugin;
+import io.github.thebusybiscuit.slimefun4.implementation.items.magical.staves.StormStaff;
+import io.github.thebusybiscuit.slimefun4.utils.LoreBuilder;
+import io.github.thebusybiscuit.slimefun4.utils.PatternUtils;
+import me.mrCookieSlime.Slimefun.Lists.RecipeType;
+import me.mrCookieSlime.Slimefun.Objects.Category;
+import me.mrCookieSlime.Slimefun.api.SlimefunItemStack;
+
+/**
+ * This class represents an item with a limited number of uses.
+ * When the item runs out of "uses", it breaks.
+ *
+ * @author Linox
+ * @author Walshy
+ * @author TheBusyBiscuit
+ * @author martinbrom
+ *
+ * @see StormStaff
+ */
+public abstract class LimitedUseItem extends SimpleSlimefunItem<ItemUseHandler> {
+
+    private final NamespacedKey defaultUsageKey = new NamespacedKey(SlimefunPlugin.instance(), "uses_left");
+    private int maxUseCount = -1;
+
+    @ParametersAreNonnullByDefault
+    public LimitedUseItem(Category category, SlimefunItemStack item, RecipeType recipeType, ItemStack[] recipe) {
+        super(category, item, recipeType, recipe, null);
+
+        addItemHandler(getItemHandler());
+    }
+
+    /**
+     * Returns the number of times this item can be used.
+     *
+     * @return The number of times this item can be used.
+     */
+    public final int getMaxUseCount() {
+        return maxUseCount;
+    }
+
+    /**
+     * Sets the maximum number of times this item can be used.
+     * The number must be greater than zero.
+     *
+     * @param count The maximum number of times this item can be used.
+     * @return The {@link LimitedUseItem} for chaining of setters
+     */
+    public final @Nonnull LimitedUseItem setMaxUseCount(int count) {
+        Validate.isTrue(count > 0, "The maximum use count must be greater than zero!");
+
+        maxUseCount = count;
+        return this;
+    }
+
+    /**
+     * Returns the {@link NamespacedKey} under which will the amount of uses left stored.
+     *
+     * @return The {@link NamespacedKey} to store/load the amount of uses
+     */
+    protected @Nonnull NamespacedKey getStorageKey() {
+        return defaultUsageKey;
+    }
+
+    @Override
+    public void register(@Nonnull SlimefunAddon addon) {
+        if (getMaxUseCount() < 1) {
+            warn("The use count has not been configured correctly. It needs to be at least 1. The Item was disabled.");
+        } else {
+            super.register(addon);
+        }
+    }
+
+    @ParametersAreNonnullByDefault
+    protected void damageItem(Player p, ItemStack item) {
```

Comment:
Not sure if `damage` is the right word here. But cannot think of a better alternative either.

Reply:
Could be called `use` but thats no better

Reply:
How about `subtractOneFromOurUsesLeftAttribute()` :kappa: